### PR TITLE
Feature/batch raw assign (depends on #189)

### PR DESCRIPTION
**Pull request by bmcdanie**
add capability to assign pointers directly into batch (skip use of view).

we lose some safety checks but the performance gains are huge. before/after callgraphs will be posted for comparison.

### CODE REVIEW

**Comment by bmcdanie** [comment]
note to self: failure caused by structured binding outlined by openmp function

File: src/batch.hpp [hunk 1]
```diff
@@ -128,6 +129,11 @@ void kronmult_to_batch_sets(
     PDE<P> const &pde);
 
 template<typename P>
```
Comment (grahamlopez) [inline src/batch.hpp]: add a quick description of this overload

File: src/batch.cpp [hunk 2]
```diff
@@ -422,6 +432,16 @@ static void kron_base(fk::matrix<P, mem_type::view, resource::device> const &A,
   batches[2].assign_entry(y_view, batch_offset);
 }
 
+// same function, high performance version
```
Comment (grahamlopez) [inline src/batch.cpp]: "// same function, unsafe version"

File: src/batch.cpp [hunk 3]
```diff
@@ -538,6 +559,75 @@ void kronmult_to_batch_sets(
   batches[pde.num_dims - 1][2].assign_entry(y_view, batch_offset);
 }
 
+template<typename P>
+void kronmult_to_batch_sets(std::vector<P *> const &A, P *const x, P *const y,
```
Comment (grahamlopez) [inline src/batch.cpp]: give us a summary of what makes this version unsafe

Comment (grahamlopez) [reply src/batch.cpp]: also, let's rename to `unsafe_kronmult_to_batch_sets()`

File: src/batch_tests.cpp [hunk 4]
```diff
@@ -143,6 +143,37 @@ TEMPLATE_TEST_CASE_SIG("batch", "[batch]",
...
+
+      test.assign_raw(second_v.data(), 1);
+      ptr_list = test.get_list();
+      REQUIRE(ptr_list[0] == first_v.data());
+      REQUIRE(ptr_list[1] == second_v.data());
+      REQUIRE(ptr_list[2] == third_v.data());
+      REQUIRE(ptr_list[0] == test(0));
+      REQUIRE(ptr_list[1] == test(1));
+      REQUIRE(ptr_list[2] == test(2));
+    }
```
Comment (grahamlopez) [inline src/batch_tests.cpp]: no tests for the `unsafe_kronmult_to_batchsets()`?

File: src/batch.hpp (hunk 1 above)
Comment (bmcdanie) [reply src/batch.hpp]: done

File: src/batch.cpp (hunk 2 above)
Comment (bmcdanie) [reply src/batch.cpp]: done

File: src/batch.cpp (hunk 3 above)
Comment (bmcdanie) [reply src/batch.cpp]: done

File: src/batch_tests.cpp (hunk 4 above)
Comment (bmcdanie) [reply src/batch_tests.cpp]: added. replicated safe section and shimmed calls into the unsafe version - the test refactor #193 will clean these up - I'll rebase that once this is merged.

**Pull request merged**